### PR TITLE
[VL] Pass file size and modification time in split 

### PR DESCRIPTION
**Pull request by acvictor**
## What changes were proposed in this pull request?

Support was made in Velox to pass file properties while creating a file handle for reading. Previously a separate call was made to fetch file properties from remote storage. By passing this information down to Velox in the split we save on one network call per file open.

https://github.com/facebookincubator/velox/issues/9314

## How was this patch tested?

Existing UTs

### CODE REVIEW

**Comment by acvictor** [comment]
@FelixYBW can you please review this? Thank you!

File: shims/common/src/main/scala/org/apache/gluten/sql/shims/SparkShims.scala [hunk 1]
```diff
@@ -207,6 +207,11 @@ trait SparkShims {
 
   def attributesFromStruct(structType: StructType): Seq[Attribute]
 
+  // Spark 3.3 and later only have file size and modification time in PartitionedFile
+  def getFileSizeAndModificationTime(file: PartitionedFile): (Option[Long], Option[Long]) = {
+    (None, None)
+  }
+
```
Comment (Yohahaha) [inline shims/common/src/main/scala/org/apache/gluten/sql/shims/SparkShims.scala:214]: we'd better keep pure abstract method in this file, just like other declaration.

**Review by Yohahaha — DISMISSED** [review]
LGTM, leave one comments.

File: shims/common/src/main/scala/org/apache/gluten/sql/shims/SparkShims.scala (hunk 1 above)
Comment (acvictor) [reply shims/common/src/main/scala/org/apache/gluten/sql/shims/SparkShims.scala]: Updated, thank you!

**Comment by acvictor** [comment]
> LGTM, leave one comments.

Thank you for the review!

**Review by Yohahaha — APPROVED** [review]
thank you!

**Comment by FelixYBW** [comment]
The PR broken the S3 reader

**Pull request merged**